contrib/gonum/plotter.go

```diff
@@ -0,0 +1,155 @@
+package gonum
+
+import (
+	"strconv"
+
+	"github.com/tobgu/qframe"
+)
+
+// LabelFunc returns a string representation of
+// the value in row i.
+type LabelFunc func(i int) string
+
+// LabelOfString returns a StringView compatible LabelFunc
+func LabelOfString(view qframe.StringView) LabelFunc {
+	return func(i int) string {
+		return *view.ItemAt(i)
+	}
+}
+
+// LabelOfEnum returns a EnumView compatible LabelFunc
+func LabelOfEnum(view qframe.EnumView) LabelFunc {
+	return func(i int) string {
+		return *view.ItemAt(i)
+	}
+}
+
+// LabelOfFloat returns a FloatView compatible LabelFunc
+// fmt determines the float format when creating a string
+func LabelOfFloat(fmt byte, view qframe.FloatView) LabelFunc {
+	return func(i int) string {
+		return strconv.FormatFloat(view.ItemAt(i), fmt, -1, 64)
+	}
+}
+
+// LabelOfInt returns an IntView compatible LabelFunc
+func LabelOfInt(view qframe.IntView) LabelFunc {
+	return func(i int) string {
+		return strconv.FormatInt(int64(view.ItemAt(i)), 64)
+	}
+}
+
+// LabelOfBool returns a BoolView compatible LabelFunc
+func LabelOfBool(view qframe.BoolView) LabelFunc {
+	return func(i int) string {
+		return strconv.FormatBool(view.ItemAt(i))
+	}
+}
+
+// Labeller implements the Labeller interface
+// defined in gonum.org/v1/plot/plotter. It accepts
+// any of the predefined LabelFunc methods in this
+// package or a custom function many be specified.
+type Labeller struct {
```

Comment:
perhaps add a `plotter_test.go` file where we check we actually implement the correct interface?

```go
package gonum_test

var (
    _ plotter.Labeller = (*gonum.Labeller)(nil)
)
```

Reply:
Added this in my most recent commit